crates/noirc_frontend/src/parser/errors.rs

```diff
@@ -147,7 +147,7 @@ impl chumsky::Error<Token> for ParserError {
             self.reason = other.reason;
         }
 
-        assert_eq!(self.span, other.span);
+        // assert_eq!(self.span, other.span);
```

Comment:
Commented out line

Reply:
I've replaced it with a commit to merge the spans since otherwise we were running into the assert. I'm not exactly sure why this PR caused us to start hitting the assert but merging the spans should be correct since we're in the merge function.